### PR TITLE
Fixed #32602 -- Clarified wording of TestCase class.

### PR DESCRIPTION
**Pull request by faishal882**
fixes ticket_32602

## what was issue
Wording was not clear between test case and test case class in documentation and some comments in code .

## solution
Updated docs and also code comment

-   [x] Do the tests still pass

### CODE REVIEW

File: django/test/runner.py [hunk 1]
```diff
@@ -495,9 +495,9 @@ def __init__(
 
     def run(self, result):
         """
-        Distribute test cases across workers.
+        Distribute test case classes across workers.
 
-        Return an identifier of each test case with its result in order to use
+        Return an identifier of each test case class with its result in order to use
```
Comment (felixxm) [inline django/test/runner.py:500]: Changed docstrings and comments should be wrapped at 79 chars.

File: django/test/runner.py [hunk 2]
```diff
@@ -1198,8 +1198,8 @@ def reorder_tests(tests, classes, reverse=False, shuffler=None):
         yield from reorder_test_bin(tests, shuffler=shuffler, reverse=reverse)
 
 
-def partition_suite_by_case(suite):
-    """Partition a test suite by test case, preserving the order of tests."""
+def partition_suite_by_test_case_class(suite):
```
Comment (felixxm) [inline django/test/runner.py:1201]: This PR should be limited to docs changes  (possibly comments/docstrings). It's not worth changing function names.

**Comment by shangxiao** [comment]
I don't think we should make these changes as the exact definitions of testing terminology differs from person to person and if these were accepted then someone else will come along in the future and submit another PR to revert it or change it to something else. 🤷‍♂️

imho the best thing to do is simply align "test case" with the class "TestCase".

**Review by felixxm — COMMENTED** [review]
@faishal882 Thanks :+1: I left comments.

File: django/test/testcases.py [hunk 3]
```diff
@@ -1193,7 +1193,7 @@ def connections_support_transactions(aliases=None):
 
 class TestData:
     """
-    Descriptor to provide TestCase instance isolation for attributes assigned
+    Descriptor to provide TestCase class instance isolation for attributes assigned
```
Comment (felixxm) [inline django/test/testcases.py:1196]: `TestCase` is a class, this change is unnecessary.

File: django/test/testcases.py [hunk 4]
```diff
@@ -251,7 +251,7 @@ def _remove_databases_failures(cls):
     def __call__(self, result=None):
         """
         Wrapper around default __call__ method to perform common Django test
-        set up. This means that user-defined Test Cases aren't required to
+        set up. This means that user-defined Test Case classes aren't required to
```
Comment (felixxm) [inline django/test/testcases.py:254]:
```suggestion
        set up. This means that user-defined TestCases aren't required to
```

**Comment by felixxm** [comment]
> imho the best thing to do is simply align "test case" with the class "TestCase".

`TestCase` is not the only option. IMO, precisioning `test case` to `test case class` in docs are not harm.

File: django/test/runner.py [hunk 5]
```diff
@@ -934,9 +934,9 @@ def build_suite(self, test_labels=None, **kwargs):
         suite = self.test_suite(all_tests)
 
         if self.parallel > 1:
-            subsuites = partition_suite_by_case(suite)
-            # Since tests are distributed across processes on a per-TestCase
-            # basis, there's no need for more processes than TestCases.
+            subsuites = partition_suite_by_test_case_class(suite)
+            # Since tests are distributed across processes on a per-TestCase class
+            # basis, there's no need for more processes than TestCase classes.
```
Comment (felixxm) [inline django/test/runner.py:939]: These changes are unnecessary, please revert.

**Comment by faishal882** [comment]
Thanks @felixxm  for your suggestion, i have made changes as requested

**Review by felixxm — COMMENTED** [review]
@faishal882 Thanks :+1: Welcome aboard :boat: 

I pushed more edits.

**Comment by faishal882** [comment]
Thanks @felixxm  for your guidance, this is my first pull request in django which has been merged excited to contibute more

**Pull request merged**